### PR TITLE
Add cmdlets for managing Backend Entity in ApiManagement

### PR DESCRIPTION
**Pull request by solankisamir**
## Description
Add Cmdlets for Managing the Backend Entity in ApiManagement.

- New-AzureRmApiManagementBackend
- Get-AzureRmApiManagementBackend
- Set-AzureRmApiManagementBackend
- Remove-AzureRmApiManagementBackend

Below cmdlets only create In-Memory Object:

- New-AzureRmApiManagementBackendCredential
- New-AzureRmApiManagementBackendProxy

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementBackend.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.New, Constants.ApiManagementBackend)]
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementBackend.cs]: This needs to support ShouldProcess

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        public PsApiManagementBackendCredential Credential { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Proxy Server details to be used while sending request to the Backend. This parameter is optional.")]
+        public PsApiManagementBackendProxy Proxy { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs]: ?? why from pipeline?

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.Set, Constants.ApiManagementBackend)]
+    [OutputType(typeof(PsApiManagementBackend))]
+    public class SetAzureApiManagementBackend : AzureApiManagementCmdletBase
+    {
+        [Parameter(
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs]: With this many named parameters, the strong suggestion is to add a new parameter set that that accepts a PSApiManagementBackEnd as one of the parameters, and use ValueFromPipeline there instead.  This is not *required*, but it is a suggestion.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "If specified then instance of " +
+                          "Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models.PsApiManagementBackend type " +
+                          " representing the modified backend will be written to output.")]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteApiManagementCmdlet()
+        {
+            Client.BackendSet(
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs]: Where is the SHouldProcess implementation?

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Microsoft.Azure.Commands.ApiManagement.ServiceManagement.dll-help.Help.pshproj [hunk 5]
```diff
@@ -5894,7 +5894,7 @@ New-AzureRmApiManagementOperation –Context $apimContext –ApiId $apiId –Ope
       </ParamSets>
       <Syntax>
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Microsoft.Azure.Commands.ApiManagement.ServiceManagement.dll-help.Help.pshproj]: This is no longe rused for help, please remove.  Instead, follow the instructions here:  https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md for creating and using the platyPS tool for markdown files.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.Set, Constants.ApiManagementBackend)]
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs]: needs to support SHoudlProcess

File: tools/StaticAnalysis/Exceptions/SignatureIssues.csv [hunk 7]
```diff
@@ -112,6 +112,12 @@
...
+"Microsoft.Azure.Commands.ApiManagement.ServiceManagement.dll","Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands.RemoveAzureApiManagementBackend","Remove-AzureRmApiManagementBackend","2","8210","Remove-AzureRmApiManagementBackend does not have a Force parameter but the cmdlet verb 'Remove' indicates that it may perform destructive actions under certain circumstances. Consider whether the cmdlet should have a Force parameter anduse ShouldContinue under some circumstances. ","Consider wether the cmdlet should have a Force parameter and use ShouldContinue under some circumstances. "
+"Microsoft.Azure.Commands.ApiManagement.ServiceManagement.dll","Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands.SetAzureApiManagementBackend","Set-AzureRmApiManagementBackend","1","8100","Set-AzureRmApiManagementBackend Does not support ShouldProcess but the cmdlet verb Set indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
```
Comment (markcowl) [inline tools/StaticAnalysis/Exceptions/SignatureIssues.csv]: Set- needs to support SHouldProcess, as noted

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementBackend.cs (hunk 1 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementBackend.cs]: #Fixed

---
In reply to: [104577049](https://github.com/Azure/azure-powershell/pull/3586#discussion_r104577049) [](ancestors = 104577049)

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs (hunk 2 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs]: #Fixed

---
In reply to: [104577220](https://github.com/Azure/azure-powershell/pull/3586#discussion_r104577220) [](ancestors = 104577220)

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs (hunk 4 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs]: #Fixed

---
In reply to: [104577504](https://github.com/Azure/azure-powershell/pull/3586#discussion_r104577504) [](ancestors = 104577504)

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs (hunk 6 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs]: #Fixed

---
In reply to: [104577790](https://github.com/Azure/azure-powershell/pull/3586#discussion_r104577790) [](ancestors = 104577790)

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Microsoft.Azure.Commands.ApiManagement.ServiceManagement.dll-help.Help.pshproj (hunk 5 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Microsoft.Azure.Commands.ApiManagement.ServiceManagement.dll-help.Help.pshproj]: Generated documentation using platyPS for new cmdlets

**Comment by solankisamir** [comment]
On demand run [1428](http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1428/)

**Pull request merged**